Simplified the import form

JevgenijVisockij opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplifies import form
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Part of #16482
| How to test?  | Simplifying Advanced Parameters -> Import form. Everything must look the same with one exception: blue help box replaced with help text under input.

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/21243)
<!-- Reviewable:end -->

JevgenijVisockij committed:
Simplified the import form

JevgenijVisockij committed:
Removed one field so it is part of simplified content

JevgenijVisockij committed:
Simplified entity to use form_row

JevgenijVisockij committed:
CS fixer fixes

JevgenijVisockij committed:
Removed placeholder for entity field

JevgenijVisockij committed:
Added back the required false

matks commented on src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportType.php:
```diff
@@ -64,17 +68,57 @@ public function buildForm(FormBuilderInterface $builder, array $options)
+            ])
+            ->add('multiple_value_separator', TextType::class, [
+                'label' => $this->trans('Multiple value separator', 'Admin.Advparameters.Feature'),
+                'help' => $this->trans('e.g. ', 'Admin.Advparameters.Help') . ' Blouse; red.jpg, blue.jpg, green.jpg; 129.90',
```
So `Blouse; red.jpg, blue.jpg, green.jpg; 129.90` stays not-translated, even if for example the BO is translated in kanjis ブラウス ?
I guess it was the same previously ?

JevgenijVisockij replied:
![image](https://user-images.githubusercontent.com/47781757/95992380-8d810d80-0e36-11eb-9c79-06f6ac5f001d.png)
Yeah

Progi1984 replied:
Ping @Julievrz

Julievrz replied:
That's a good question. In my opinion, it should be translated but maybe there was a reason why it was not @LouiseBonnard ?

LouiseBonnard replied:
I agree, the rest of the help texts should be localized. I tried with the page displayed in German and it appears untranslated:

<img width="843" alt="Capture d’écran 2020-10-29 à 11 48 51" src="https://user-images.githubusercontent.com/32565890/97558835-f2af3400-19dc-11eb-81c8-0f88d88ab7d1.png">

Julievrz replied:
Thanks Louise for the confirmation 😄 

So, "Blouse; red.jpg, blue.jpg, green.jpg; 129.90" as well as "1; Blouse; 129.90; 5" should localized in Admin.Advparameters.Help

LouiseBonnard replied:
Why not localize it as a whole, like `e.g. Blouse; red.jpg, blue.jpg, green.jpg; 129.90`?

Julievrz replied:
Yes you're right. If it is the same for the code then it is better to do it that way. 👍

matks reviewed (dismissed):
PR seems OK, just asked one question 😉

khouloudbelguith commented:
Hi @JevgenijVisockij,

I found some issues:

1. All fields are required => NOK
![image](https://user-images.githubusercontent.com/16067358/96241338-4ca60780-0fa2-11eb-8506-0d810989403b.png)

2. There is no separator between the `Multiple value separator` block and `Delete all` block
**With PR**
![image](https://user-images.githubusercontent.com/16067358/96241694-d229b780-0fa2-11eb-8f3e-9b85b46a7c38.png)
**Without PR**
![image](https://user-images.githubusercontent.com/16067358/96241823-fa191b00-0fa2-11eb-86e8-644894fad404.png)

3. When we want to import `Catgeories`, this option is displayed "Use product reference as key" => NOK (It is always displayed => it should be displayed only when we want to import `Products` or `Combinations`
![image](https://user-images.githubusercontent.com/16067358/96242097-5e3bdf00-0fa3-11eb-8870-60bd420fb246.png)
4. When we want to import `Combinations`, this option is displayed "Skip thumbnails regeneration" => NOK (It is always displayed => it should be displayed only when we want to import `Categories` or `Products` or `Brands` or `Suppliers` or` Store Contacts`
5. There is no help text under the `Use product reference as key` field
**With PR**
![image](https://user-images.githubusercontent.com/16067358/96242325-a529d480-0fa3-11eb-970e-3808f0c08e48.png)
**Without PR**
![image](https://user-images.githubusercontent.com/16067358/96242513-d30f1900-0fa3-11eb-8026-b590944a1ca0.png)

6. If you try to import without selecting a file, you have an alert
**With PR**
![image](https://user-images.githubusercontent.com/16067358/96243944-b7a50d80-0fa5-11eb-823b-aa76abab71ab.png)
**Without PR**
![image](https://user-images.githubusercontent.com/16067358/96243990-c8ee1a00-0fa5-11eb-9259-bed58a285f59.png)

Thanks!

JevgenijVisockij committed:
Fixed issue with product reference not hidden when it should be

JevgenijVisockij committed:
Fixed issue with certain fields displaying when they shouldn't

JevgenijVisockij committed:
Added missing help text for reference

JevgenijVisockij committed:
Fixed translation domain

JevgenijVisockij committed:
Removed uneeded space

JevgenijVisockij committed:
Fixed problem with error appearing about extra field where it shouldn't

JevgenijVisockij committed:
Added required false to choice type

JevgenijVisockij committed:
Added back the separator

JevgenijVisockij committed:
Moved useless variable closer to the forn

JevgenijVisockij commented:
@khouloudbelguith  fixed the issues, but some fields still have the red * next to them as they are actually required.

This is ready for code reviews.

JevgenijVisockij committed:
Removed useless variable as found a better way to disable rendering of the button

JevgenijVisockij committed:
CS fixer fixes

Julievrz commented:
> This pull request seems to contain new translation strings. I have summarized them below to ease up review:
> 
> * `Admin.Advparameters.Notification`
>   
>   * [ ]  [`The locale must be installed `](https://github.com/PrestaShop/PrestaShop/pull/21245/files#diff-504663c882a367661b111a8c50c0d377R76)
> 
> (Note: this is an automated message, but answering it will reach a real human)

@atomiix This wording doesn't seem new ?

Julievrz commented on src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportType.php:
```diff
@@ -64,17 +70,62 @@ public function buildForm(FormBuilderInterface $builder, array $options)
+            ])
+            ->add('sendemail', SwitchType::class, [
+                'label' => $this->trans('Send notification email', 'Admin.Advparameters.Feature'),
+                'help' => $this->trans('Sends an email to let you know your import is complete. It can be useful when handling large files, as the import may take some time.', 'Admin.Advparameters.Help'),
```
```suggestion
                'help' => $this->trans('Receive an email when the import is complete. It can be useful when handling large files, as the import may take some time.', 'Admin.Advparameters.Help'),
```

Julievrz commented on src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportType.php:
```diff
@@ -64,17 +70,62 @@ public function buildForm(FormBuilderInterface $builder, array $options)
+                    'class' => 'js-force-ids-form-group',
+                ],
+                'label' => $this->trans('Force all ID numbers', 'Admin.Advparameters.Feature'),
+                'help' => $this->trans('If you enable this option, your imported items\' ID number will be used as is. If you do not enable this option, the imported ID numbers will be ignored, and PrestaShop will instead create auto-incremented ID numbers for all the imported items.', 'Admin.Advparameters.Help'),
```
```suggestion
                'help' => $this->trans('Enable this option to keep your imported items\’ ID number as is. Otherwise, PrestaShop will ignore them and create auto-incremented ID numbers.', 'Admin.Advparameters.Help'),
```

Julievrz commented:
I made some suggestions to make the wording more concise :)

JevgenijVisockij committed:
ADded block form import form

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportType.php

Made wording more concise

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

JevgenijVisockij commented:
@Julievrz  implemented your suggestions.

> > This pull request seems to contain new translation strings. I have summarized them below to ease up review:
> > 
> > * `Admin.Advparameters.Notification`
> >   
> >   * [x]   [`The locale must be installed `](https://github.com/PrestaShop/PrestaShop/pull/21245/files#diff-504663c882a367661b111a8c50c0d377R76)
> > 
> > (Note: this is an automated message, but answering it will reach a real human)
> 
> @atomiix This wording doesn't seem new ?

And regarding this I am guessing it's the same case of me fixing something after noticing Bot's comments. I guess in the future I will write the comment "I fixed this and that translation after bot complained about them" to reduce confusion.

Julievrz commented:
> @Julievrz implemented your suggestions.
> 
> > > This pull request seems to contain new translation strings. I have summarized them below to ease up review:
> > > 
> > > * `Admin.Advparameters.Notification`
> > >   
> > >   * [x]    [`The locale must be installed `](https://github.com/PrestaShop/PrestaShop/pull/21245/files#diff-504663c882a367661b111a8c50c0d377R76)
> > > 
> > > (Note: this is an automated message, but answering it will reach a real human)
> > 
> > 
> > @atomiix This wording doesn't seem new ?
> 
> And regarding this I am guessing it's the same case of me fixing something after noticing Bot's comments. I guess in the future I will write the comment "I fixed this and that translation after bot complained about them" to reduce confusion.

Thank you @JevgenijVisockij ! I think there are still 2 of my suggestions left before I can put the wording ✔️ label 😃

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportType.php

Updated wording

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

JevgenijVisockij commented:
> > @Julievrz implemented your suggestions.
> > > > This pull request seems to contain new translation strings. I have summarized them below to ease up review:
> > > > 
> > > > * `Admin.Advparameters.Notification`
> > > >   
> > > >   * [x]     [`The locale must be installed `](https://github.com/PrestaShop/PrestaShop/pull/21245/files#diff-504663c882a367661b111a8c50c0d377R76)
> > > > 
> > > > (Note: this is an automated message, but answering it will reach a real human)
> > > 
> > > 
> > > @atomiix This wording doesn't seem new ?
> > 
> > 
> > And regarding this I am guessing it's the same case of me fixing something after noticing Bot's comments. I guess in the future I will write the comment "I fixed this and that translation after bot complained about them" to reduce confusion.
> 
> Thank you @JevgenijVisockij ! I think there are still 2 of my suggestions left before I can put the wording heavy_check_mark label smiley

Oh sorry I missed one. I commited this https://github.com/PrestaShop/PrestaShop/pull/21245#discussion_r514179487, but I couldn't find the second one though :thinking: 

The only other suggestion I can see is https://github.com/PrestaShop/PrestaShop/pull/21245#discussion_r514207720 but it was already commited.

Julievrz commented on src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportType.php:
```diff
@@ -64,17 +70,62 @@ public function buildForm(FormBuilderInterface $builder, array $options)
+            ])
+            ->add('separator', TextType::class, [
+                'label' => $this->trans('Field separator', 'Admin.Advparameters.Feature'),
+                'help' => $this->trans('e.g. ', 'Admin.Advparameters.Help') . ' 1; Blouse; 129.90; 5',
```
```suggestion
                'help' => $this->trans('e.g. 1; Blouse; 129.90; 5', 'Admin.Advparameters.Help'),
```

Julievrz commented on src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportType.php:
```diff
@@ -64,17 +70,62 @@ public function buildForm(FormBuilderInterface $builder, array $options)
+            ])
+            ->add('multiple_value_separator', TextType::class, [
+                'label' => $this->trans('Multiple value separator', 'Admin.Advparameters.Feature'),
+                'help' => $this->trans('e.g. ', 'Admin.Advparameters.Help') . ' Blouse; red.jpg, blue.jpg, green.jpg; 129.90',
```
```suggestion
                'help' => $this->trans('e.g. Blouse; red.jpg, blue.jpg, green.jpg; 129.90 ', 'Admin.Advparameters.Help'), 
```

Julievrz commented:
> > > @Julievrz implemented your suggestions.
> > > > > This pull request seems to contain new translation strings. I have summarized them below to ease up review:
> > > > > 
> > > > > * `Admin.Advparameters.Notification`
> > > > >   
> > > > >   * [x]      [`The locale must be installed `](https://github.com/PrestaShop/PrestaShop/pull/21245/files#diff-504663c882a367661b111a8c50c0d377R76)
> > > > > 
> > > > > (Note: this is an automated message, but answering it will reach a real human)
> > > > 
> > > > 
> > > > @atomiix This wording doesn't seem new ?
> > > 
> > > 
> > > And regarding this I am guessing it's the same case of me fixing something after noticing Bot's comments. I guess in the future I will write the comment "I fixed this and that translation after bot complained about them" to reduce confusion.
> > 
> > 
> > Thank you @JevgenijVisockij ! I think there are still 2 of my suggestions left before I can put the wording heavy_check_mark label smiley
> 
> Oh sorry I missed one. I commited this [#21245 (comment)](https://github.com/PrestaShop/PrestaShop/pull/21245#discussion_r514179487), but I couldn't find the second one though 🤔
> 
> The only other suggestion I can see is [#21245 (comment)](https://github.com/PrestaShop/PrestaShop/pull/21245#discussion_r514207720) but it was already commited.

It is about the examples being localized in Admin.Advparameters.Help, I added the suggestion. 😊

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportType.php

Localized the examples for separators.

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

JevgenijVisockij committed:
Update src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportType.php

Localized the examples for separators.

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

JevgenijVisockij commented:
> > > > @Julievrz implemented your suggestions.
> > > > > > This pull request seems to contain new translation strings. I have summarized them below to ease up review:
> > > > > > 
> > > > > > * `Admin.Advparameters.Notification`
> > > > > >   
> > > > > >   * [x]       [`The locale must be installed `](https://github.com/PrestaShop/PrestaShop/pull/21245/files#diff-504663c882a367661b111a8c50c0d377R76)
> > > > > > 
> > > > > > (Note: this is an automated message, but answering it will reach a real human)
> > > > > 
> > > > > 
> > > > > @atomiix This wording doesn't seem new ?
> > > > 
> > > > 
> > > > And regarding this I am guessing it's the same case of me fixing something after noticing Bot's comments. I guess in the future I will write the comment "I fixed this and that translation after bot complained about them" to reduce confusion.
> > > 
> > > 
> > > Thank you @JevgenijVisockij ! I think there are still 2 of my suggestions left before I can put the wording heavy_check_mark label smiley
> > 
> > 
> > Oh sorry I missed one. I commited this [#21245 (comment)](https://github.com/PrestaShop/PrestaShop/pull/21245#discussion_r514179487), but I couldn't find the second one though thinking
> > The only other suggestion I can see is [#21245 (comment)](https://github.com/PrestaShop/PrestaShop/pull/21245#discussion_r514207720) but it was already commited.
> 
> It is about the examples being localized in Admin.Advparameters.Help, I added the suggestion. blush

Sorry, I forgot about that comment and was searching specifically for suggestions. Just commited the suggestions, thanks for adding them!

JevgenijVisockij committed:
CS fixer fix

khouloudbelguith commented:
Hi @JevgenijVisockij,

As discussed with @LouiseBonnard.
I found an issue following these steps:
1. Go to BO > Advanced Parameters > Import page
If you select `Categories`, `Products`, `Customers`, `Addresses`, `Brands`, `Suppliers`, `Alias` or `Stores contact`:
The option "Force all ID numbers" is displayed, in the help text under input, we have:

`Enable this option to keep your imported items\’ ID number as is. Otherwise, PrestaShop will ignore them and create auto-incremented ID numbers.`

The  `\` is not accepted.
![image](https://user-images.githubusercontent.com/16067358/99791134-77492a00-2b25-11eb-880d-13d6cdbfb85c.png)

Thank you!

JevgenijVisockij committed:
Removed not needed slash

JevgenijVisockij commented:
> Hi @JevgenijVisockij,
> 
> As discussed with @LouiseBonnard.
> I found an issue following these steps:
> 
>     1. Go to BO > Advanced Parameters > Import page
>        If you select `Categories`, `Products`, `Customers`, `Addresses`, `Brands`, `Suppliers`, `Alias` or `Stores contact`:
>        The option "Force all ID numbers" is displayed, in the help text under input, we have:
> 
> 
> `Enable this option to keep your imported items\’ ID number as is. Otherwise, PrestaShop will ignore them and create auto-incremented ID numbers.`
> 
> The `\` is not accepted.
> ![image](https://user-images.githubusercontent.com/16067358/99791134-77492a00-2b25-11eb-880d-13d6cdbfb85c.png)
> 
> Thank you!

Good catch, thanks!

I have removed it.

khouloudbelguith commented:
Hi @JevgenijVisockij,

It is ok :heavy_check_mark: 
Just I found this [bug](https://github.com/PrestaShop/PrestaShop/issues/22372), but it is not related to this PR.

Thanks!

atomiix commented:
Thanks @JevgenijVisockij & @khouloudbelguith !
